### PR TITLE
Add configmap and secret to runner pod

### PR DESCRIPTION
**Pull request by machacekondra**
This PR add support to pass secret and config map to the chaos-runner
pod.

Signed-off-by: Ondra Machacek <machacek.ondra@gmail.com>

**What this PR does / why we need it**:
Fixes: https://github.com/litmuschaos/litmus/issues/2174

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [] Fixes #2174
-   [ ] Labelled this PR & related issue with `documentation` tag
-   [ ] PR messages has document related information
-   [ ] Labelled this PR & related issue with `breaking-changes` tag
-   [ ] PR messages has breaking changes related information
-   [ ] Labelled this PR & related issue with `requires-upgrade` tag
-   [ ] PR messages has upgrade related information
-   [ ] Commit has unit tests
-   [ ] Commit has integration tests

### CODE REVIEW

**Comment by machacekondra** [comment]
This PR depends on: https://github.com/litmuschaos/elves/pull/32

**Comment by ksatchit** [comment]
@machacekondra thanks for this PR! Have merged the dependent PR in elves - would you like to pull the vendor changes

**Comment by ksatchit** [comment]
> @machacekondra thanks for this PR! Have merged the dependent PR in elves - would you like to pull the vendor changes. Also can you add the deepcopy changes (upon `operator-sdk generate k8s`) as applicable.

@machacekondra - there is some discussion around maintaining vendor v/s removal going on - however, having said that the plan is to go with the current approach for 1.9.0

**Comment by machacekondra** [comment]
> @machacekondra thanks for this PR! Have merged the dependent PR in elves - would you like to pull the vendor changes. Also can you add the deepcopy changes (upon `operator-sdk generate k8s`) as applicable.

Added, thanks!

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 1]
```diff
@@ -220,20 +220,22 @@ func getChaosRunnerENV(cr *litmuschaosv1alpha1.ChaosEngine, aExList []string, Cl
...
-
 // newGoRunnerPodForCR defines a new go-based Runner Pod
 func newGoRunnerPodForCR(engine *chaosTypes.EngineInfo) (*corev1.Pod, error) {
+	engine.VolumeOpts.VolumeOperations(engine.Instance.Spec.Components.Runner.ConfigMaps, engine.Instance.Spec.Components.Runner.Secrets)
+
 	containerForRunner := container.NewBuilder().
 		WithEnvsNew(getChaosRunnerENV(engine.Instance, engine.AppExperiments, analytics.ClientUUID)).
 		WithName("chaos-runner").
 		WithImage(engine.Instance.Spec.Components.Runner.Image).
+		WithVolumeMountsNew(engine.VolumeOpts.VolumeMounts).
```
Comment (ispeakc0de) [inline pkg/controller/chaosengine/chaosengine_controller.go:238]: As we are returning an error inside the builder function if `VolumeMounts` is nil. Do we have to put this field as optional, allow only if `engine.VolumeOpts.VolumeMounts` is not nil?

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 2]
```diff
@@ -248,13 +250,14 @@ func newGoRunnerPodForCR(engine *chaosTypes.EngineInfo) (*corev1.Pod, error) {
...
 
-    podForRunner := pod.NewBuilder().
+	podForRunner := pod.NewBuilder().
 		WithName(engine.Instance.Name + "-runner").
 		WithNamespace(engine.Instance.Namespace).
 		WithAnnotations(engine.Instance.Spec.Components.Runner.RunnerAnnotation).
 		WithLabels(getChaosRunnerLabels(engine.Instance)).
 		WithServiceAccountName(engine.Instance.Spec.ChaosServiceAccount).
 		WithRestartPolicy("OnFailure").
+		WithVolumeBuilders(engine.VolumeOpts.VolumeBuilders).
```
Comment (ispeakc0de) [inline pkg/controller/chaosengine/chaosengine_controller.go:260]: same as above

**Comment by ksatchit** [comment]
Branch updated to add fix/workarounds for bdd tests (causing the failure in circle-ci)

**Review by ispeakc0de — APPROVED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**